### PR TITLE
Docs about Passing JSON to the View

### LINKED ISSUE #1900: Add to docs/ core the way to encode JSON via hsx
```
import qualified Text.Blaze.Html.Renderer.Text as Renderer
import Text.Blaze.Html (Html)
import qualified Data.Text.Lazy as TL

instance ToJSON Html where
    toJSON html = String (TL.toStrict $ Renderer.renderHtml html)
    
    
<div data-my-encoded-json={encode someJsonValue}></div>
```

### PR DESCRIPTION
**Pull request by amitaibu**
fixes #1900

### CODE REVIEW

File: Guide/view.markdown [hunk 1]
```diff
@@ -181,6 +181,64 @@ Now the `company` variable can be used to read the current user's company across
...
+
+postsToJson :: [Post] -> Value
+postsToJson posts =
+    posts
+        |> fmap (\post -> object
+            [ "id" .= post.id
+            , "title" .= post.title
+            , "body" .= post.body
+            ])
+        |> toJSON
```
Comment (mpscholten) [inline Guide/view.markdown:207]: I'd actually move this to the view file as well and then pass the `posts` list directly to the view

Comment (amitaibu) [reply Guide/view.markdown:207]: I can surely change, but before here's my reasoning: The controller should normally do the heavy lifting. Often it's not as simple as the `postsToJson` function. 

In the beginning my Views would have lots of logic, until I realized it's better to think of them inside the controller and let the view print or filter if needed.

Anyway, either way works, so let me know what you prefer :)

Comment (mpscholten) [reply Guide/view.markdown:207]: > Often it's not as simple as the postsToJson function

Agree, in those cases I'd also try to keep it in the controller (or even the Helpers)

But in this specific case it IMO really belongs to the view. Especially when the JS code for decoding that json is in the file as well

Comment (amitaibu) [reply Guide/view.markdown]: done

**Review by mpscholten — APPROVED** [review]
Thanks 👍

**Pull request merged**